python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -365,10 +365,20 @@ def _delete_artifact(self, location: Location) -> None:
             raise
         log.debug("Successfully deleted file: %s", location.uri)
 
-    def addStoredItemInfo(self, refs: Iterable[DatasetRef], infos: Iterable[StoredFileInfo]) -> None:
+    def addStoredItemInfo(
+        self, refs: Iterable[DatasetRef], infos: Iterable[StoredFileInfo], insert_mode: str = "insert"
```

Comment:
Maybe use `Literal["insert", ...]` instead of `str`?

Reply:
Should I use an enum? This is an internal interface so maybe that is better.

Reply:
👍 Enum is probably more natural here.